authentication-ldap/src/test/java/com/synopsys/integration/alert/authentication/ldap/action/LDAPTestActionTest.java

```diff
@@ -0,0 +1,195 @@
+
+        AlertProperties alertProperties = new MockAlertProperties();
+        FilePersistenceUtil filePersistenceUtil = new FilePersistenceUtil(alertProperties, new Gson());
+        EncryptionUtility encryptionUtility = new EncryptionUtility(alertProperties, filePersistenceUtil);
+        MockLDAPConfigurationRepository mockLDAPConfigurationRepository = new MockLDAPConfigurationRepository();
+        ldapConfigAccessor = new LDAPConfigAccessor(encryptionUtility, mockLDAPConfigurationRepository);
+
+        ldapTestAction = new LDAPTestAction(authorizationManager, authenticationDescriptorKey, ldapConfigurationValidator, ldapManager, ldapConfigAccessor);
+
+        ldapConfigModel.setManagerDn("managerDn");
+        ldapConfigModel.setManagerPassword("managerPassword");
+        ldapConfigModel.setServerName("serverName");
+        ldapConfigModel.setIsManagerPasswordSet(true);
+        ldapConfigModel.setEnabled(false);
+    }
+
+    @Test
+    void testAuthenticationValidationFails() {
+        LDAPConfigModel emptyConfigModel = new LDAPConfigModel();
+        LDAPConfigTestModel ldapConfigTestModel = new LDAPConfigTestModel(emptyConfigModel, testUser, testPass);
+        ActionResponse<ValidationResponseModel> validationResponseModelActionResponse = ldapTestAction.testAuthentication(ldapConfigTestModel);
+
+        assertEquals(HttpStatus.OK, validationResponseModelActionResponse.getHttpStatus());
+
+        // asserts to show we failed validation
+        assertTrue(validationResponseModelActionResponse.getContent().isPresent());
+        assertTrue(validationResponseModelActionResponse.getContent().get().hasErrors());
+        assertTrue(validationResponseModelActionResponse.getContent().get().getErrors().size() > 0);
+        assertEquals("There were problems with the configuration", validationResponseModelActionResponse.getContent().get().getMessage());
+    }
+
+    @Test
+    void testAuthenticationValidationSuccess() {
+        LDAPConfigTestModel ldapConfigTestModel = new LDAPConfigTestModel(ldapConfigModel, testUser, testPass);
+        ActionResponse<ValidationResponseModel> validationResponseModelActionResponse = ldapTestAction.testAuthentication(ldapConfigTestModel);
+
+        assertEquals(HttpStatus.OK, validationResponseModelActionResponse.getHttpStatus());
+
+        // assert to show we passed validation
+        assertTrue(validationResponseModelActionResponse.getContent().isPresent());
+        assertEquals(0, validationResponseModelActionResponse.getContent().get().getErrors().size());
+
+        // assert to show testConfigModelContent() caught an exception
+        assertTrue(validationResponseModelActionResponse.getContent().get().hasErrors());
+        assertEquals("LDAP Test Configuration failed. Error creating LDAP authenticator", validationResponseModelActionResponse.getContent().get().getMessage());
+    }
```

Comment:
After the isPresent() check we should call get() and get the ValidationResponseModel. That should  be used to cut down on the number of calls on the optional.

This applies to the tests below as well.

Reply:
The other thing you can do is use `orElseThrow(()-> new AssertionError("Expected resonponse content not found"))`